app/grandchallenge/components/utils/virtualenvs.py

```diff
@@ -0,0 +1,22 @@
+import subprocess
+from pathlib import Path
+
+
+def run_script_in_venv(*, venv_location, python_script, args=None):
+    """
+    Runs a Python script as a subprocess in an existing isolated virtual environment.
+
+    Returns the result of the process.
+    """
+    venv_path_activate = Path(venv_location).resolve() / "bin" / "activate"
+    python_script = Path(python_script).resolve()
+    args = " ".join(args or [])
+
+    command = f"source {venv_path_activate} && python {python_script} {args}"
```

Comment:
Unsure, but I think that this should use `shlex` rather than f-string building: https://docs.python.org/3/library/shlex.html#shlex.quote

Reply:
Good point, that will make the pathing space proof and add some security to the additional args. It's an internal function but I'd rather be overly cautious than sorry later.

(Haha. 'Space proof' has a nice ring to it!)